src/Swarm/TUI/Inventory/Sorting.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+-- |
+-- Module      :  Swarm.TUI.Inventory.Sorting
+-- Copyright   :  Brent Yorgey
+-- Maintainer  :  byorgey@gmail.com
+--
+-- SPDX-License-Identifier: BSD-3-Clause
+module Swarm.TUI.Inventory.Sorting (
+  InventorySortOptions (..),
+  InventorySortDirection (..),
+  InventorySortOrder (..),
+  cycleSortOrder,
+  cycleSortDirection,
+  sortInventory,
+  renderSortMethod,
+) where
+
+import Control.Lens (view)
+import Data.List (sortBy)
+import Data.Ord (comparing)
+import Data.Text qualified as T
+
+import Algorithms.NaturalSort (sortKey)
+
+import Swarm.Game.Entity as E
+
+data InventorySortDirection
+  = Ascending
+  | Descending
+  deriving (Enum, Bounded, Eq)
+
+data InventorySortOrder
+  = ByNaturalAlphabetic
+  | ByQuantity
+  | ByType
+  deriving (Enum, Bounded, Eq)
+
+data InventorySortOptions = InventorySortOptions InventorySortDirection InventorySortOrder
+
+renderSortMethod :: InventorySortOptions -> T.Text
+renderSortMethod (InventorySortOptions direction order) =
+  T.unwords [prefix, label]
+ where
+  prefix = case direction of
+    Ascending -> "↑"
+    Descending -> "↓"
+  label = case order of
+    ByNaturalAlphabetic -> "name"
+    ByQuantity -> "count"
+    ByType -> "type"
+
+succCyclic :: (Enum a, Bounded a, Eq a) => a -> a
```

Comment:
We already have this function in `Swarm.Util.cycleEnum`.  (`succCyclic` is probably a better name, we can definitely rename it if we want, but in any case we shouldn't duplicate it.)